spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableScanRDD.scala

```diff
@@ -325,10 +325,11 @@ class CassandraTableScanRDD[R] private[connector](
   }
 
   private def fetchTokenRange(
-    session: Session,
+    scanner: Scanner,
     range: CqlTokenRange[_, _],
-    inputMetricsUpdater: InputMetricsUpdater,
-    scanExecuteMethod: Statement => (Iterator[Row], CassandraRowMetadata)): Iterator[R] = {
+    inputMetricsUpdater: InputMetricsUpdater): Iterator[R] = {
+
+    val session = scanner.getSession()
```

Comment:
it looks like `createStatement` could be internal for Scanner. This way we wouldn't have to expose session. Session would be a private thing